yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java

```diff
@@ -0,0 +1,66 @@
+package com.yoti.api.client.spi.remote;
+
+import static com.yoti.api.client.spi.remote.call.YotiConstants.DEFAULT_CHARSET;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.SYMMETRIC_CIPHER;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+
+import java.io.UnsupportedEncodingException;
+import java.security.Key;
+import java.security.PrivateKey;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.crypto.spec.SecretKeySpec;
+
+import com.yoti.api.client.ActivityDetails;
+import com.yoti.api.client.Profile;
+import com.yoti.api.client.ProfileException;
+import com.yoti.api.client.spi.remote.call.Receipt;
+import com.yoti.api.client.spi.remote.util.Decryption;
+
+public class ActivityDetailsFactory {
```

Comment:
Why is this Factory public? 
It really shouldn't be used from client code.

Reply:
Fixed that one, and a few others where public access is not required.

This does force us to keep components in the same package, which is not ideal, but if we want to restrict access I don't see another way of doing it.